# Linear Regression:Real Estate Analysis

In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

C:\Users\manis\Anaconda3\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)


In [3]:
house_price = pd.read_excel("Linear Regression.xlsx")

In [11]:
house_price.head(10)

price  sqft_living  bedrooms  bathrooms  floors
0   221900         1180         3       1.00     1.0
1   538000         2570         3       2.25     2.0
2   180000          770         2       1.00     1.0
3   604000         1960         4       3.00     1.0
4   510000         1680         3       2.00     1.0
5  1225000         5420         4       4.50     1.0
6   257500         1715         3       2.25     2.0
7   291850         1060         3       1.50     1.0
8   229500         1780         3       1.00     1.0
9   323000         1890         3       2.50     2.0

In [4]:
house_price.isna().sum()

price          0
sqft_living    0
bedrooms       0
bathrooms      0
floors         0
dtype: int64

In [5]:
house_price.dtypes

price            int64
sqft_living      int64
bedrooms         int64
bathrooms      float64
floors         float64
dtype: object

___Dependent variable is Price and Independent variable is sqft_living___

In [13]:
x = house_price.iloc[:,:1]
y = house_price.iloc[:,1:2]

In [16]:
train_x,test_x,train_y,test_y = train_test_split(x,y,test_size=0.2,random_state = 0)
lin_reg = LinearRegression()

In [17]:
y

sqft_living
0             1180
1             2570
2              770
3             1960
4             1680
...            ...
21608         1530
21609         2310
21610         1020
21611         1600
21612         1020

[21613 rows x 1 columns]

In [18]:
lin_reg.fit(train_x,train_y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [19]:
lin_reg.coef_

array([[0.00174499]])

In [20]:
lin_reg.intercept_

array([1134.23538784])

In [22]:
y_pred = lin_reg.predict(test_x)
y_pred

array([[1652.49798531],
       [3887.83262287],
       [2115.09534083],
       ...,
       [1779.79514519],
       [1657.73296104],
       [2139.26347879]])

In [27]:
from sklearn.metrics import r2_score,mean_squared_error
r2_value = r2_score(test_y,y_pred)
rmse = np.sqrt(mean_squared_error(test_y,y_pred))
print("R-Square error",r2_value)
print("Root Mean squre error",rmse)

R-Square error 0.4813925927530154
Root Mean squre error 647.3715598616461


___sqft_feet shows 48.139 impact on deciding the house price___

___dependent variable is price and Independent variable is bedrooms___

In [28]:
y1 = house_price.iloc[:,2:3]
y1

bedrooms
0             3
1             3
2             2
3             4
4             3
...         ...
21608         3
21609         4
21610         2
21611         3
21612         2

[21613 rows x 1 columns]

In [29]:
x1_train,x1_test,y1_train,y1_test = train_test_split(x,y1,test_size=0.2,random_state = 2)


In [30]:
lin_reg1 = LinearRegression()
lin_reg1.fit(x1_train,y1_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [31]:
lin_reg1.coef_

array([[7.70125623e-07]])

In [32]:
lin_reg1.intercept_

array([2.95061418])

In [37]:
y1_pred=lin_reg1.predict(x1_test)
y1_pred

array([[3.51665652],
       [3.83625865],
       [3.22054321],
       ...,
       [3.25670061],
       [3.23941129],
       [3.11080031]])

In [38]:
rmse1 = np.sqrt(mean_squared_error(y1_test,y1_pred))
r2_value1 = r2_score(y1_test,y1_pred)
print("R-Square error",r2_value1)
print("Root Mean squre error",rmse1)


R-Square error 0.10933217805863649
Root Mean squre error 0.8736548931683022


___bedrooms variable shows 10.93% impact on deciding the house price___

___dependent variable is price and independent variable is bathrooms___

In [39]:
y2 = house_price.iloc[:,3:4]
y2

bathrooms
0           1.00
1           2.25
2           1.00
3           3.00
4           2.00
...          ...
21608       2.50
21609       2.50
21610       0.75
21611       2.50
21612       0.75

[21613 rows x 1 columns]

In [41]:
x2_train,x2_test,y2_train,y2_test = train_test_split(x,y2,test_size = 0.2,random_state = 2)
lin_reg2 = LinearRegression()
lin_reg2.fit(x2_train,y2_train)


LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [43]:
y2_pred = lin_reg2.predict(x2_test)
y2_pred

array([[2.32779742],
       [2.78473399],
       [1.90444293],
       ...,
       [1.95613732],
       [1.9314187 ],
       [1.74754302]])

In [44]:
rmse2 = np.sqrt(mean_squared_error(y2_test,y2_pred))
r2_value2 = r2_score(y2_test,y2_pred)
print("R-Square error",r2_value2)
print("Root Mean squre error",rmse2)


R-Square error 0.28127401846250877
Root Mean squre error 0.6591011174526225


___Bathrooms features shows 28.127% impact on deciding the house pricing___

___Dependent variable is price and Independent variable is floors___

In [46]:
y3 = house_price.iloc[:,4:]
y3

floors
0         1.0
1         2.0
2         1.0
3         1.0
4         1.0
...       ...
21608     3.0
21609     2.0
21610     2.0
21611     2.0
21612     2.0

[21613 rows x 1 columns]

In [47]:
x3_train,x3_test,y3_train,y3_test = train_test_split(x,y3,test_size = 0.2,random_state = 2)
lin_reg3 = LinearRegression()
lin_reg3.fit(x3_train,y3_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [48]:
lin_reg3.coef_

array([[3.73194413e-07]])

In [49]:
lin_reg3.intercept_

array([1.29281316])

In [50]:
y3_pred = lin_reg3.predict(x3_test)
y3_pred

array([[1.56711105],
       [1.72198673],
       [1.4236178 ],
       ...,
       [1.44113928],
       [1.43276106],
       [1.3704376 ]])

In [51]:
rmse3 = np.sqrt(mean_squared_error(y3_test,y3_pred))
r2_value3 = r2_score(y3_test,y3_pred)
print("R-Square error",r2_value3)
print("Root Mean squre error",rmse3)


R-Square error 0.07374135097127821
Root Mean squre error 0.5222402255396246


___Floors variable shows 7.37% impact on deciding the house price___